run-dljc.sh

```diff
@@ -100,179 +116,107 @@ fi
-      TMP="${DLJC_CMD} -- ${BUILD_CMD}"
-      DLJC_CMD="${TMP}"
-      
-      # ensure the project is clean before invoking DLJC
-      eval ${CLEAN_CMD} < /dev/null
-
-      echo ${DLJC_CMD}
-      
-      eval ${DLJC_CMD} < /dev/null
-      
-      if [[ $? -eq 124 ]]; then
-          echo "dljc timed out for ${REPO_NAME}"
-          echo "dljc timed out for ${REPO_NAME}" > ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-	  USABLE="no"
-      else 
-          if [ -f dljc-out/wpi.log ]; then
-              cp dljc-out/wpi.log ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-              echo "${REPO},${HASH}" >> ../../../${OUTDIR}-results/interesting-results.csv
-	      USABLE="yes"
-          else
-              # if this last run was under Java 11, try to run
-              # under Java 8 instead
-              if [ "${JAVA_HOME}" = "${JAVA11_HOME}" ]; then
-                  export JAVA_HOME="${JAVA8_HOME}"
-                  echo "couldn't build using Java 11; trying Java 8"
-                  configure_and_exec_dljc
-              else
-                  echo "dljc could not run the build successfully" > ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-		  USABLE="no"
-              fi
-          fi
-      fi
-  fi
-  export JAVA_HOME="${JAVA11_HOME}"
-}
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
 
 while IFS='' read -r line || [ "$line" ]
 do    
     REPOHASH=${line}
 
-    REPO=`echo ${REPOHASH} | awk '{print $1}'`
-    HASH=`echo ${REPOHASH} | awk '{print $2}'`
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
 
-    REPO_NAME=`echo ${REPO} | cut -d / -f 5`
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
+    REPO_NAME_HASH="${REPO_NAME}-${HASH}"
 
-    # need a layer in the file structure that prevents
-    # two repos with the same name from colliding
-    if [ ! -d ${REPO_NAME}-${HASH} ]; then
-        mkdir ${REPO_NAME}-${HASH}
-    fi
+    # Use repo name and hash, but not owner.  We want
+    # repos that are different but have the same name to be treated
+    # as different repos, but forks with the same content to be skipped.
+    # TODO: consider just using hash, to skip hard forks?
+    mkdir -p "${REPO_NAME_HASH}"
 
-    pushd ${REPO_NAME}-${HASH}
+    pushd "${REPO_NAME_HASH}" || exit 5
     
-    if [ ! -d ${REPO_NAME} ]; then
-        # this environment variable prevents git from prompting for username/password
-        # if the repository no longer exists
-        GIT_TERMINAL_PROMPT=0 git clone ${REPO}
+    if [ ! -d "${REPO_NAME}" ]; then
+        # this environment variable prevents git from prompting for
+	# username/password if the repository no longer exists
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
     else
-        rm -rf ${REPO_NAME}/dljc-out
+        rm -rf "${REPO_NAME}/dljc-out"
     fi
 
     # if the above clone command failed for whatever reason, create an
```

Comment:
Rather than doing this, why not use a `continue` statement to not execute all the commands below?  Or if you don't want to use `continue`, you could put all the rest of the loop within an `if` statement.

Reply:
I'm not sure whether this comment is referring to removing the `dljc-out` directory or the line following this comment.

If it's the former, the behavior is intentional: the script repeats the experiment, but removes the cache of files created by the previous round. This allows the experimenter to re-use the same set of downloaded repositories to run multiple experiments, e.g. with bugfixes to a checker. That's a behavior that I want to preserve.

If it's the latter, you're right and I will do that.